File: Algoritmo Ordenação/selection-sort/java/selection-sort.java
import java.util.Arrays;

class SelectionSort {
    public static void selectionSort(int[] v) {
        int indiceDoMenor;

        for (int i = 0; i < v.length - 1; i++) {
            indiceDoMenor = i;

            for (int j = i + 1; j < v.length; j++) {
                if (v[j] < v[indiceDoMenor]) {
                    indiceDoMenor = j;
                }
            }

            if (i != indiceDoMenor) {
                int aux = v[i];
                v[i] = v[indiceDoMenor];
                v[indiceDoMenor] = aux;
            }
        }

        System.out.printf("Vetor ordenado => %s", Arrays.toString(v));
    }

    public static void main(String[] args) {
        int[] v1 = { 4, 7, 5, 2, 10, 8, 1, 6, 3 };

        selectionSort(v1);
    }
}

File: Algoritmos/Algoritmo Ordenação/selection-sort/typescript/selection-sort.ts
var vetor: Array<number> = [ 5, 3, 2, 4, 7, 1, 0, 6 ];

function selectionSort(vetor: Array<number>): Array<number> {
  let indiceDoMenor: number;

  for (let i = 0; i < vetor.length - 1; i++) {
    indiceDoMenor = i;

    for (let j = i + 1; j < vetor.length; j++) {
      if (vetor[j] < vetor[indiceDoMenor]) {
        indiceDoMenor = j;
      }
    }

    if (i !== indiceDoMenor) {
      [vetor[i], vetor[indiceDoMenor]] = [vetor[indiceDoMenor], vetor[i]];
    }
  }

  return vetor;
}

console.log("SELECTION SORT 1 TS => ", selectionSort(vetor));
